src/SimulationRunner.cc

```diff
@@ -480,14 +483,40 @@ void SimulationRunner::AddSystemToRunner(SystemInternal _system)
 {
   this->systems.push_back(_system);
 
-  if (_system.preupdate)
-    this->systemsPreupdate.push_back(_system.preupdate);
+  // Decide if the system is going to be added to the list of systems
+  // based on its current role.
+  // TO-DO: doing this does not disable running the configure method of the
+  // system. Verify if running that and not running the system later do not
+  // cause issues with the simulation
+  if (this->networkMgr)
+  {
+    if (this->networkMgr->IsSecondary())
+    {
+      igndbg << "Adding system to Secondary." << std::endl;
+      if (_system.postupdate)
+        this->systemsPostupdate.push_back(_system.postupdate);
+    }
+    else
+    {
+      igndbg << "Adding system to primary." << std::endl;
+      if (_system.preupdate)
+        this->systemsPreupdate.push_back(_system.preupdate);
 
-  if (_system.update)
-    this->systemsUpdate.push_back(_system.update);
+      if (_system.update)
+        this->systemsUpdate.push_back(_system.update);
+    }
+  }
+  else
+  {
+    if (_system.preupdate)
+      this->systemsPreupdate.push_back(_system.preupdate);
 
-  if (_system.postupdate)
-    this->systemsPostupdate.push_back(_system.postupdate);
+    if (_system.update)
```

Comment:
May checking here and below that is not secondary instead of repeating lines in the if